### PR TITLE
Update sbt 1.10.1 + GitHub Actions

### PR DESCRIPTION
**Pull request by eed3si9n**
Ref https://github.com/sbt/sbt-jupiter-interface/pull/59

**Problem/Solution**
This is a minimum update to bump the overall build to the latest sbt and GitHub Actions via sbt-github-actions.

### CODE REVIEW

**Comment by mkurz** [comment]
Thanks, will take a look asap.

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,11 +1,4 @@
...
-)
-
-resolvers += Resolver.typesafeIvyRepo("releases")
-
-addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.7")
-
-// For sbt 0.13.x (upto sbt-sonatype 2.3)
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")
+addSbtPlugin("com.github.sbt" % "sbt-release" % "1.1.0")
+addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "3.9.21")
```
Comment (eed3si9n) [inline project/plugins.sbt:2]: Likely I'll remove these in favor of the usual sbt-ci-release later.

File: build.sbt [hunk 2]
```diff
@@ -84,62 +87,55 @@ lazy val root = (project in file("."))
...
     s"junit.platform.version=${Versions.junitPlatform}\n" +
     s"junit.jupiter.version=${Versions.junitJupiter}\n" +
     s"junit.vintage.version=${Versions.junitVintage}\n"
   IO.write(file, content)
   Seq(file)
 }
+
+ThisBuild / homepage := Some(url("https://github.com/maichler/sbt-jupiter-interface"))
+ThisBuild / licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0"))
+ThisBuild / organization := "com.github.sbt"
```
Comment (eed3si9n) [inline build.sbt:124]: Fixes https://github.com/sbt/sbt-jupiter-interface/issues/64

**Comment by eed3si9n** [comment]
Merging this now.

**Comment by mkurz** [comment]
It's interesting you named the org `com.github.sbt.junit`, that's the first time I see this pattern in the sbt organization. Usually everything is just `com.github.sbt` % `sbt-...`

**Comment by eed3si9n** [comment]
> It's interesting you named the org `com.github.sbt.junit`, that's the first time I see this pattern in the sbt organization.

Yea, it's because this repo has a pure Java jupiter-interface module, so I figured it would be nice if we can make the Java package match the organization name.

**Pull request merged**